src/core.rs

```diff
@@ -118,6 +120,10 @@ impl Core {
         let userpass_module = UserPassModule::new(self);
         self.module_manager.add_module(Arc::new(RwLock::new(Box::new(userpass_module))))?;
 
+        // add credential module: approle
+        let approle_module = AppRoleModule::new(self);
+        self.module_manager.add_module(Arc::new(RwLock::new(Box::new(approle_module))))?;
```

Comment:
Should this just panic directly?

Reply:
> Should this just panic directly?

There is no direct `panic`. The `?` operator returns immediately if there is an error without executing the subsequent operations; otherwise, `unwrap` is auto used to get the value inside `Ok`.

Reply:
No, I mean this should panic directly instead of returning to upper caller. Since if the add_module fails, the whole program should be terminated.